src/services/feedProcessors/MultiFeedProcessor.ts

```diff
@@ -1,50 +1,98 @@
 import {inject, injectable} from 'inversify';
 import {Logger} from 'winston';
 
+import {ByBitSpotFetcher, BinancePriceMultiFetcher, CoingeckoPriceMultiFetcher} from '../fetchers/index.js';
+import UniswapV3MultiFetcher from '../dexes/uniswapV3/UniswapV3MultiFetcher.js';
+import {SovrynPriceFetcher} from '../dexes/sovryn/SovrynPriceFetcher.js';
 import {FeedFetcher} from '../../types/Feed.js';
-import {mergeArrays} from '../../utils/collections.js';
-import CoingeckoMultiProcessor from './CoingeckoMultiProcessor.js';
-import CryptoCompareMultiProcessor from './CryptoCompareMultiProcessor.js';
-import UniswapV3MultiProcessor from './UniswapV3MultiProcessor.js';
-import SovrynMultiProcessor from '../dexes/sovryn/SovrynMultiProcessor.js';
-import ByBitMultiProcessor from './ByBitMultiProcessor.js';
-import BinanceMultiProcessor from './BinanceMultiProcessor.js';
+import {FetcherName, FetcherResult} from '../../types/fetchers.js';
 
 @injectable()
-export default class MultiFeedProcessor {
+export default class MultiFeedProcessorNew {
+  @inject(CoingeckoPriceMultiFetcher) coingeckoPriceFetcher!: CoingeckoPriceMultiFetcher;
+  @inject(BinancePriceMultiFetcher) binancePriceFetcher!: BinancePriceMultiFetcher;
+  @inject(ByBitSpotFetcher) byBitSpotPriceFetcher!: ByBitSpotFetcher;
+  @inject(SovrynPriceFetcher) sovrynPriceFetcher!: SovrynPriceFetcher;
+  @inject(UniswapV3MultiFetcher) uniswapV3PriceFetcher!: UniswapV3MultiFetcher;
   @inject('Logger') logger!: Logger;
-  @inject(CoingeckoMultiProcessor) coingeckoMultiProcessor!: CoingeckoMultiProcessor;
-  @inject(CryptoCompareMultiProcessor) cryptoCompareMultiProcessor!: CryptoCompareMultiProcessor;
-  @inject(UniswapV3MultiProcessor) uniswapV3MultiProcessor!: UniswapV3MultiProcessor;
-  @inject(SovrynMultiProcessor) sovrynMultiProcessor!: SovrynMultiProcessor;
-  @inject(ByBitMultiProcessor) byBitMultiProcessor!: ByBitMultiProcessor;
-  @inject(BinanceMultiProcessor) binanceMultiProcessor!: BinanceMultiProcessor;
 
   async apply(feedFetchers: FeedFetcher[]): Promise<unknown[]> {
     if (!feedFetchers.length) return [];
 
-    let response: unknown[] = [];
-    response.length = feedFetchers.length;
+    this.logger.debug(`[MultiFeedProcessorNew] feedFetchers ${JSON.stringify(feedFetchers)}`);
+
+    const inputMap = new Map<
```

Comment:
- please create a type for it
- I think fetcher can have a type not any
- I think we can remove `mergeArrays` method

later on, when we switch to DB, all this code hopefully will be removed.

Reply:
addressed